hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -214,6 +214,18 @@ protected void putTransientInternal(Object key, Data value, long ttl, TimeUnit t
         }
     }
 
+    @Override
+    protected InternalCompletableFuture<Data> putIfAbsentAsyncInternal(Object key, Data valueData,
+                                                                       long ttl, TimeUnit ttlUnit,
+                                                                       long maxIdle, TimeUnit maxIdleUnit) {
+        key = toNearCacheKeyWithStrategy(key);
+        try {
+            return super.putIfAbsentAsyncInternal(key, valueData, ttl, ttlUnit, maxIdle, maxIdleUnit);
+        } finally {
+            invalidateNearCache(key);
```

Comment:
I think we don't need to invalidate keys in cases where puts are not succeed.
Actually, if we don't keep null values in near cache (idk this detail), it seems to me that there will never be a need for invalidation, but at least I'm sure we shouldn't do this invalidation in cases where the update doesn't happen.
I also saw the non-async version, it also invalidates but I think we should discuss it with someone more knowledgeable in this near cache.

Reply:
This matter is tricky. If the operation is retried, the return value can be wrong. Consider: key `k` isn't present. We do `putIfAbsent('k', v)`. The operation puts the value, updates the backup, but before the response is sent, the member crashes. The caller doesn't receive the response, so retries with the new key owner. This time, the operation does nothing because the key isn't absent, and returns the old value.

The unnecessary invalidation doesn't hurt as much as non-invalidated near cache would. But OTOH, the bad return value is much much less likely than the unnecessary invalidation...

Reply:
Thanks for the explanation, I didn't think about that before. If it were up to me, I wouldn't retry such non-idempotent operation invocations, but this discussion is beyond the scope of PR.

Reply:
@ufukyilmaz good thinking. 
As @viliam-durina mentioned, here it is ok to invalidate near cache for all ending conditions due to distributed system shortcomings.